py/server/deephaven/experimental/iceberg.py

```diff
@@ -250,3 +251,80 @@ def adapter_aws_glue(
     except Exception as e:
         raise DHError(e, "Failed to build Iceberg Catalog Adapter") from e
 
+
+def adapter(
+        properties: Dict[str, str],
+        name: Optional[str] = None
+) -> IcebergCatalogAdapter:
+    """
+    Create an Iceberg catalog adapter for an Iceberg catalog created from configuration properties. These properties
+    map to the Iceberg catalog Java API properties and are used to create the catalog and file IO implementations.
+
+    The minimal set of properties required to create an Iceberg catalog are the following:
+    - `catalog-impl` or `type` - the Java catalog implementation to use. When providing `catalog-impl`, the
+            implementing Java class should be provided (e.g. `org.apache.iceberg.rest.RESTCatalog` or
+            `org.apache.iceberg.aws.glue.GlueCatalog`. Choices for `type` include `hive`, `hadoop`, `rest`, `glue`,
+            `nessie`, `jdbc`.
+    - `uri` - the URI of the catalog
+    - `io-impl` - the Java file IO implementation to use. Common choices are: `org.apache.iceberg.aws.s3.S3FileIO`
+            and `org.apache.iceberg.hadoop.HadoopFileIO`
+
+    Other common properties include:
+    - `warehouse` - the location of the data warehouse.
+    - `client.region` - the region of the AWS client.
+    - `s3.access-key-id` - the S3 access key for reading files.
+    - `s3.secret-access-key` - the S3 secret access key for reading files.
+    - `s3.endpoint` - the S3 endpoint to connect to.
+
+    Example usage #1 - REST catalog with an S3 backend (using MinIO):
+    ```
+    from deephaven.experimental import s3, iceberg
+
+    properties = {
+        "type" : "rest",
+        "uri" : "http://rest:8181",
+        "io-impl" : "org.apache.iceberg.aws.s3.S3FileIO",
```

Comment:
Can this be generally inferred from other properties (ie, `s3.<x.`, or `org.apache.iceberg.io.ResolvingFileIO`)?

Reply:
See comment below, how much inference do we need to perform on our users' behalf?

Reply:
Well, I somewhat am going off of some minimal iceberg examples. https://iceberg.apache.org/docs/1.6.0/jdbc/#java-api shows

```java
Map<String, String> properties = new HashMap<>();
properties.put(CatalogProperties.CATALOG_IMPL, JdbcCatalog.class.getName());
properties.put(CatalogProperties.URI, "jdbc:mysql://localhost:3306/test");
properties.put(CatalogProperties.WAREHOUSE_LOCATION, "s3://warehouse/path");
Configuration hadoopConf = new Configuration();
JdbcCatalog catalog = CatalogUtil.buildIcebergCatalog("test_jdbc_catalog", properties, hadoopConf);
```

(somewhat edited for brevity). But above, they are _not_ explicitly setting `org.apache.iceberg.CatalogProperties.FILE_IO_IMPL`; if `CatalogUtil` doesn't require it, do we still need it?